File: PlayGround/List/List.js
// make string "Hello World" and string "11223344"
let str1 = "Hello World";
let int1 = "11223344";

console.log("Str:" +str1+ " Int:"+int1+"\n");

// turn string into list
let list_str1 = str1.split("");
let list_int1 = int1.split("");

console.log(list_str1);
console.log(list_int1);

// turn list string into list int
list_int1 = list_int1.map(Number);
console.log(list_int1);

// remove duplicate
list_int1 = [...new Set(list_int1)];
console.log(list_int1);

// turn list into string
console.log(list_str1.join(""));

// check index space
let ind_space = list_str1.indexOf(" ");
console.log(ind_space);

// remove space in list
list_str1.splice(ind_space, 1);
console.log(list_str1);

// swap list
[list_str1[4], list_str1[5]] = [list_str1[5], list_str1[4]]
console.log(list_str1);

// make another list
let list_int2 = [6,7,8,9,10];


// merge 2 list with same type
list_int1 = list_int1.concat(list_int2);
console.log(list_int1);

// check if 'k' inside list
console.log(list_int1.includes(9));

// cut list into 2 left and right
mid = list_str1.length/2
left = list_str1.splice(0, mid);
right =list_str1.splice(-mid);

console.log(left);
console.log(right);

// make hash map
let fistHashMap = new Map([
    [1, "first"],
    [2, "second"],
    [3, "third"]
])

// check if n in keys
console.log(fistHashMap.has(1));

// get second value
console.log(fistHashMap.get(1));

// check if n in values
console.log(Array.from(fistHashMap.values()).includes("second"));

for(let [key, value] of fistHashMap){
    console.log(key+" = "+value);
}

File: PlayGround/List/List.dart
import 'dart:collection';

List<String> swap(int x, int y, list_str) {
  var tmp = list_str[x];
  list_str[x] = list_str[y];
  list_str[y] = tmp;

  return list_str;
}

void main() {
  // make string "Hello World" and string "1122334455"
  String str1 = "Hello World";
  String int1 = "1122334455";

  print(str1);
  print(int1);

  // turn string into list
  List<String> list_str = str1.split("");
  print(list_str);

  List<String> list_int = int1.split("");
  print(list_int);

  // turn list string into list int
  List<int> arr_int = list_int.map(int.parse).toList();
  print(arr_int);
  print(arr_int.runtimeType);

  // turn list into string
  print(list_str.join());
  // remove duplicate
  arr_int = arr_int.toSet().toList();
  print(arr_int);
  // check if 'space' inside list
  print(list_str.contains(" "));

  // check index space
  int index_space = list_str.indexOf(" ");
  print(index_space);

  // remove space in list
  list_str.removeAt(index_space);
  print(list_str);

  // swap list
  list_str = swap(4, 5, list_str);
  print(list_str);

  // make another list 678910
  List<int> arr_int2 = [6, 7, 8, 9, 10];

  // merge 2 list with same type
  arr_int = arr_int + arr_int2;
  print(arr_int);

  // cut list into 2 left and right
  var list_length = list_str.length;
  var mid = (list_length / 2).floor();

  List<String> left = list_str.sublist(0, mid);
  List<String> right = list_str.sublist(mid, list_length);

  print(left);
  print(right);
  // make hash map
  Map<int, String> hash_map = {1: "first", 2: "second", 3: "third"};

  // check if n in keys
  print(hash_map.containsKey(1));

  // get hash_map get value of k hash_map[]
  print(hash_map[1]);
  // check if n in values
  print(hash_map.containsValue("fourth"));

  // itterate hashmap
  hash_map.forEach((key, value) {
    print("$key = $value");
  });
}
